core/src/Revolution/Processors/Security/Role/GetList.php

```diff
@@ -90,21 +95,28 @@ public function beforeIteration(array $list)
      */
     public function prepareRow(xPDOObject $object)
     {
-        $objectArray = $object->toArray();
-        $isCoreRole = $object->get('id') === 1
-            || $object->get('id') === 2
-            || $object->get('name') === 'Super User'
-            || $object->get('name') === 'Member';
-
-        $perm = [];
-        if (!$isCoreRole) {
-            $perm[] = 'edit';
-            if ($this->canRemove) {
-                $perm[] = 'remove';
-            }
+        // Note: Role does not have a checkPolicy() method
+        $permissions = [
+            'create' => $this->canCreate,
+            'update' => $this->canUpdate,
+            'delete' => $this->canDelete
+        ];
+
+        $roleData = $object->toArray();
+        $roleName = $object->get('name');
+        $isCoreRole = $roleName === 'Super User' || $roleName === 'Member';
+
+        if ($isCoreRole) {
+            $baseKey = '_role_' . strtolower(str_replace(' ', '', $roleName)) . '_';
+            $roleData['name_trans'] = $this->modx->lexicon($baseKey . 'name');
+            $roleData['description_trans'] = $this->modx->lexicon($baseKey . 'description');
         }
-        $objectArray['perm'] = implode(' ', $perm);
 
-        return $objectArray;
+        $roleData['reserved'] = ['name' => ['Super User', 'Member']];
+        $roleData['isProtected'] = $isCoreRole ? true : false ;
```

Comment:
```suggestion
        $roleData['isProtected'] = $isCoreRole;
```

Reply:
Good catch, changed in all three GetList classes.